### PR TITLE
Support k8s_node monitored resource in Custom Metrics - Stackdriver Adapter

### PR DESCRIPTION
**Pull request by kawych**

### CODE REVIEW

**Comment by kawych** [comment]
@crassirostris @x13n @loburm 
Can one of you take a look?

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 1]
```diff
@@ -83,18 +83,65 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
...
 // with any namespace. Current implementation doesn't support root scoped metrics.
 func (p *StackdriverProvider) GetRootScopedMetricByName(groupResource schema.GroupResource, name string, metricName string) (*custom_metrics.MetricValue, error) {
-	return nil, provider.NewOperationNotSupportedError("Get root scoped metric by name")
+	if !p.translator.useNewResourceModel {
+		return nil, provider.NewOperationNotSupportedError("Get root scoped metric by name")
+	}
+	if groupResource.Resource != "Node" {
+		return nil, provider.NewOperationNotSupportedError(fmt.Sprintf("Get root scoped metric by name for resource %q", groupResource.Resource))
+	}
+	matchingNode, err := p.kubeClient.Nodes().Get(name, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (kawych) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: This is not cached. This is the same for pods, right? This is only called by HPA, so shouldn't be too often. I think this should be addressed in a separate PR if the issue is signifficant.

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 2]
```diff
@@ -83,18 +83,65 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
...
 }
 
 // GetRootScopedMetricBySelector queries Stackdriver for metrics identified by selector and not
 // associated with any namespace. Current implementation doesn't support root scoped metrics.
 func (p *StackdriverProvider) GetRootScopedMetricBySelector(groupResource schema.GroupResource, selector labels.Selector, metricName string) (*custom_metrics.MetricValueList, error) {
-	return nil, provider.NewOperationNotSupportedError("Get root scoped metric by selector")
+	if !p.translator.useNewResourceModel {
+		return nil, provider.NewOperationNotSupportedError("Get root scoped metric by selector")
+	}
+	if groupResource.Resource != "nodes" {
```
[the remark this answers is not in the document]
Comment (kawych) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: This was a mistake, done

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 3]
```diff
@@ -83,18 +83,65 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
...
+		return nil, provider.NewOperationNotSupportedError(fmt.Sprintf("Get root scoped metric by selector for resource %q", groupResource.Resource))
+	}
+	matchingNodes, err := p.kubeClient.Nodes().List(metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		return nil, err
+	}
+	result := []custom_metrics.MetricValue{}
+	for i := 0; i < len(matchingNodes.Items); i += 100 {
```
[the remark this answers is not in the document]
Comment (kawych) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: Done

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 4]
```diff
@@ -83,18 +83,65 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
...
+		stackdriverRequest, err := p.translator.GetSDReqForNodes(&v1.NodeList{Items: matchingNodes.Items[i:min(i+100, len(matchingNodes.Items))]}, metricName)
```
[the remark this answers is not in the document]
Comment (kawych) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: Done

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 5]
```diff
@@ -83,18 +83,65 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
...
+		}
+		result = append(result, slice...)
+	}
+	return &custom_metrics.MetricValueList{Items: result}, nil
 }
 
 // GetNamespacedMetricByName queries Stackdriver for metrics identified by name and associated
 // with a namespace.
 func (p *StackdriverProvider) GetNamespacedMetricByName(groupResource schema.GroupResource, namespace string, name string, metricName string) (*custom_metrics.MetricValue, error) {
+	if groupResource.Resource != "pods" {
```
[the remark this answers is not in the document]
Comment (kawych) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: Done

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 6]
```diff
@@ -107,12 +154,15 @@ func (p *StackdriverProvider) GetNamespacedMetricByName(groupResource schema.Gro
...
 		return nil, err
 	}
-	return p.translator.GetRespForPod(stackdriverResponse, groupResource, metricName, namespace, name)
+	return p.translator.GetRespForSingleObject(stackdriverResponse, groupResource, metricName, namespace, name)
 }
 
 // GetNamespacedMetricBySelector queries Stackdriver for metrics identified by selector and associated
 // with a namespace.
 func (p *StackdriverProvider) GetNamespacedMetricBySelector(groupResource schema.GroupResource, namespace string, selector labels.Selector, metricName string) (*custom_metrics.MetricValueList, error) {
+	if groupResource.Resource != "pod" {
```
[the remark this answers is not in the document]
Comment (kawych) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: Done

**Comment by crassirostris** [comment]
Please ping the PR when the failing unit test is fixed

**Comment by crassirostris** [comment]
LGTM

**Pull request merged**